### PR TITLE
Bypass `pre_upgrade` bug by reinstalling groups

### PR DESCRIPTION
**Pull request by hpeebles**

### CODE REVIEW

File: backend/canisters/group/impl/src/updates/c2c_initialize_events.rs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+use group_canister::c2c_initialize_events::{Response::*, *};
+use std::collections::HashMap;
+use types::{
+    EventIndex, GroupPermissions, MessageContentInternal, MessageIndex, PermissionRole, Role, TimestampMillis, UserId,
+};
+use utils::time::MINUTE_IN_MS;
+
+#[update_msgpack(guard = "caller_is_group_index")]
+#[trace]
+async fn c2c_initialize_events(args: Args) -> Response {
```
Comment (megrogan) [inline backend/canisters/group/impl/src/updates/c2c_initialize_events.rs:18]: Doesn't need to be async

File: backend/canisters/local_group_index/impl/src/model/local_group_map.rs [hunk 2]
```diff
@@ -34,9 +34,9 @@ impl LocalGroupMap {
         }
     }
 
-    pub fn iter(&self) -> impl Iterator<Item = (&ChatId, &LocalGroup)> {
-        self.groups.iter()
-    }
+    // pub fn iter(&self) -> impl Iterator<Item = (&ChatId, &LocalGroup)> {
```
Comment (megrogan) [inline backend/canisters/local_group_index/impl/src/model/local_group_map.rs:37]: delete me

File: backend/canisters/user_index/impl/src/lib.rs [hunk 3]
```diff
@@ -207,6 +214,33 @@ impl Data {
             }
         }
     }
+
+    fn register_super_admin_bot(&mut self, canister_id: CanisterId, username: String, now: TimestampMillis) {
```
Comment (megrogan) [inline backend/canisters/user_index/impl/src/lib.rs:218]: Why doesn't this need to be public?

Comment (hpeebles) [reply backend/canisters/user_index/impl/src/lib.rs:218]: It's in the root and modules can access function and types in their ancestors.
So things defined in `lib` only need to be `pub` if they're accessed from other packages.

File: backend/integration_tests/src/reinstall_group_tests.rs [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                    &mut env,
+                    user.principal,
+                    group_id.into(),
+                    &group_canister::unpin_message::Args {
+                        message_index: i.into(),
+                        correlation_id: 0,
+                    },
+                );
+            }
+        }
```
Comment (megrogan) [inline backend/integration_tests/src/reinstall_group_tests.rs:118]: Let's edit some messages and leave some reactions

**Review by megrogan — APPROVED** [review]

**Review by megrogan — DISMISSED** [review]

**Review by megrogan — DISMISSED** [review]

File: backend/canisters/group/impl/src/lib.rs [hunk 5]
```diff
@@ -62,6 +62,10 @@ impl RuntimeState {
         caller == self.data.group_index_canister_id || caller == self.data.local_group_index_canister_id
```
Comment (megrogan) [inline backend/canisters/group/impl/src/lib.rs:62]: Should this now be more specific?

**Review by megrogan — DISMISSED** [review]

**Review by megrogan — APPROVED** [review]

**Review by megrogan — APPROVED** [review]

**Pull request merged**